app/controllers/projects/samples/attachments_controller.rb

```diff
@@ -28,12 +28,23 @@ def create
       end
 
       def destroy
-        authorize! @project, to: :update_sample?
+        authorize! @project, to: :destroy?
+        return unless @attachment.attachable_type == 'Sample' && @attachment.attachable_id == @sample.id
```

Comment:
We should return an error to display to the user instead of just just returning

Reply:
Fixed in d9f8225c5d145849cec6a8481c7897234d4b8b2d

Reply:
Sorry Chris I made a mistake here, we should change this back to `update_sample?`.  The `destroy_sample?` authorization method only allows a member who is an owner to delete attachments, but we want to allow maintainers to delete attachments as well

Reply:
> I'm not sure how this if condition could ever be hit. The reason being that `@attachment` is queried from the `@sample`, so it should only be set if the attachment belongs to the sample. There is also no test for this condition, so that would be required if this if statement is possible to be met which I cannot see it ever happening. If the user were trying to delete an attachment that doesn't belong to the sample they would get `not_found` as defined within the `attachment` method in this controller.

Based on @ericenns comment below, I've removed the check in ffb0a839fee97a3e3eb2bf7596e329fc451758ee